core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -113,6 +113,9 @@ public Builder addPlugin(Class<? extends Plugin> pluginClass) {
          * Builds a new instance of the transport client.
          */
         public TransportClient build() {
+            if (settings.get("plugin.types") != null) {
```

Comment:
Just check the `settings` when they are set on the builder and throw an `IllegalArgumentException` there?